File: leetcode/POTD/2025-02-04/solution.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int max = 0, curr = nums[0];
        for(int i=1;i<nums.length;i++){
            if(nums[i] > nums[i-1]){
                curr += nums[i];
            } else {
                max = Math.max(max, curr);
                curr = nums[i];
            }
        }
        return Math.max(max, curr);
    }
}

File: leetcode/POTD/2025-02-04/solution.rs
impl Solution {
    pub fn max_ascending_sum(nums: Vec<i32>) -> i32 {
        let mut max_sum = 0;
        let mut curr = nums[0];

        for i in 1..nums.len() {
            if nums[i] > nums[i - 1] {
                curr += nums[i];
            } else {
                max_sum = max_sum.max(curr);
                curr = nums[i];
            }
        }

        max_sum.max(curr)
    }
}
